djoser/backends.py

```diff
@@ -0,0 +1,21 @@
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import Permission
+from django.contrib.auth.backends import ModelBackend
+from django.db.models import Exists, OuterRef, Q
+from djoser.conf import settings
+
+User = get_user_model()
+
+
+class LoginFieldBackend(ModelBackend):
+
+    def authenticate(self, request, username=None, password=None, **kwargs):
+        if username is None:
+            username = kwargs.get(settings.LOGIN_FIELD)
+        if username is None or password is None:
+            return
+        user = User.objects.filter(**kwargs).first()
+        if user and not user.check_password(password):
+            return
+        if user and user.is_active:
```

Comment:
This is good to have so djoser is compatible with Django's default auth model backend. However to be fully covered in the future I would either try to call `super().authenticate()` or at least [`self.user_can_authenticate()`](https://github.com/django/django/blob/5.0.4/django/contrib/auth/backends.py#L51-L56).

Reply:
I'd definitely release the fix in 2.3 as it's a minor security issue, but at the same time I'm not sure if these particular changes are actually needed.

IMO the changes from https://github.com/sunscrapers/djoser/commit/8f65bfff16577c7fb0f52bbabf5fb69f6809ba62 are incorrect. If the user didn't pass `authenticate`, then they shouldn't be able to login.

If you need a custom behavior, it's up to the `ModelBackend` and user manager implementation (like @hisie did). Handling it by Djoser is a shortcut.